src/exportJarFileCommand.ts

```diff
@@ -0,0 +1,337 @@
+        });
+        for (const pickDependency of pickDependencies) {
+            if (pickDependency.picked) {
+                pickedDependencies.push(pickDependency);
+            }
+        }
+        const pickBox = createPickBox("Export Jar - Determine elements", "Select the elements...", pickDependencies, true, pickSteps.length);
+        pickBox.selectedItems = pickedDependencies;
+        pickBox.onDidTriggerButton((item) => {
+            if (item === QuickInputButtons.Back) {
+                reject(InputFlowAction.back);
+                pickBox.dispose();
+            }
+        });
+        pickBox.onDidAccept(() => {
+            for (const item of pickBox.selectedItems) {
+                outClassPaths.push(item.uri);
+            }
+            resolve(outClassPaths);
+            pickBox.dispose();
+        });
+        pickBox.onDidHide(() => {
+            reject();
+            pickBox.dispose();
+        });
+        pickBox.show();
+    });
+}
+
+function createPickBox(title: string, placeholder: string, items: IJarQuickPickItem[],
+                       canSelectMany: boolean, pickStepLength: number): QuickPick<IJarQuickPickItem> {
+    const pickBox = window.createQuickPick<IJarQuickPickItem>();
+    pickBox.title = title;
+    pickBox.placeholder = placeholder;
+    pickBox.canSelectMany = canSelectMany;
+    pickBox.items = items;
+    pickBox.ignoreFocusOut = true;
+    pickBox.buttons = pickStepLength > 0 ? [(QuickInputButtons.Back)] : [];
+    return pickBox;
+}
+
+function generateDependencies(paths: string[], setUris: Set<string>, projectPath: string, isRuntime: boolean): IJarQuickPickItem[] {
+    const pickDependencies: IJarQuickPickItem[] = [];
+    for (const classpath of paths) {
+        const extName = extname(classpath);
+        const baseName = (extName === ".jar") ? basename(classpath) : classpath.substring(projectPath.length + 1);
+        const descriptionValue = (isRuntime) ? "Runtime" : "Test";
+        const typeValue = (extName === ".jar") ? "external" : "internal";
+        if (!setUris.has(classpath)) {
+            setUris.add(classpath);
+            const jarQuickPickItem: IJarQuickPickItem = {
+                label: baseName,
+                description: descriptionValue,
+                uri: classpath,
+                type: typeValue,
+                picked: isRuntime,
+            };
+            pickDependencies.push(jarQuickPickItem);
+        }
+    }
+    return pickDependencies;
+}
+
+function getName(data: MainMethodInfo) {
+    return data.name.substring(data.name.lastIndexOf(".") + 1);
+}
+
+class ClasspathResult {
+    public projectRoot: string;
+    public classpaths: string[];
+    public modulepaths: string[];
+}
+
+export class MainMethodInfo {
+    public name: string;
+    public path: string;
+}
+
+class InputFlowAction {
+    public static back = new InputFlowAction();
+}
+
+interface IJarQuickPickItem extends QuickPickItem {
+    label: string;
```

Comment:
label. description, picked are all declared in QuickPickItem , no need to redeclare